packages/react-components/src/Popup/PopupWindow.tsx

```diff
@@ -0,0 +1,90 @@
+// Copyright 2017-2021 @polkadot/react-components authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import type { PopupWindowProps, WindowProps } from './types';
+
+import React from 'react';
+import { createPortal } from 'react-dom';
+import styled, { css } from 'styled-components';
+
+import { useElementPosition } from '@polkadot/react-hooks';
+
+import { getPosition } from './utils';
+
+export const PopupWindow: React.FC<PopupWindowProps> = ({ children, position, setRef, triggerPosition }) => {
+  const windowPosition = useElementPosition(setRef);
+
+  return createPortal(
+    <Window
+      positionX={position}
+      positionY={triggerPosition.verticalPosition}
+      ref={setRef}
+      windowPosition={getPosition(triggerPosition, position, triggerPosition.verticalPosition, windowPosition)}
+    >
+      {children}
+    </Window>,
+    document.body
+  );
+};
+
+const Window = styled.div<WindowProps>`
```

Comment:
Don't use this form in styled-components, rather follow the same pattern as used elsewhere, i.e. div + className + styles on React.memo

Reply:
✅